tools/pot/openvino/tools/pot/graph/passes.py

```diff
@@ -489,6 +490,52 @@ def disconnect_fq_node(self, fq_node):
         if parent_node.type == 'Const':
             parent_node['need_shape_inference'] = True
 
+    def optimize_for_gp_hw(self, graph, target_device):
+        """
+        Removing redundant FQs before operation Add
+        for SPR(CPU) and ATS(GPU) platform
+        """
+        def _walk_for_branch(node):
+            input_node = node
+            delete_const = lambda node: ([op for op in node if op is not None and op.type != 'Const'])
+            while True:
+                input_node = get_node_inputs(input_node)
+                input_node = delete_const(input_node)
+                if len(input_node) > 1:
+                    return False
+                input_node = input_node[0]
+                if input_node.type in ['Convolution', 'GroupConvolution', 'MatMul']:
+                    return True
+
+        def _check_const_input(node):
+            input = get_node_inputs(node)[0]
+            return nu.check_const_input(input)
+
+        def delete_one_fq(inputs_node):
+            fq_1, fq_2 = inputs_node
+            if len(get_all_node_outputs(fq_1)) > 1 and len(get_all_node_outputs(fq_2)) == 1 and _check_const_input(fq_2):
+                self.disconnect_fq_node(fq_2)
+                return
+            if _walk_for_branch(fq_1) and _walk_for_branch(fq_2):
+                if np.prod(nu.get_output_shape(fq_1, 0)) >= np.prod(nu.get_output_shape(fq_2, 0)):
+                    self.disconnect_fq_node(fq_1)
+                else:
+                    self.disconnect_fq_node(fq_2)
+                return
+
+        if target_device not in ['CPU_SPR', 'GPU_ATS']:
+            return
+
+        check_is_inputs_fq = lambda node: all([op.type == 'FakeQuantize' for op in node])
+        for op in get_nodes_by_type(graph, ['Add']):
+            if not nu.check_const_input(op):
+                inputs_node = np.array(get_node_inputs(op))
```

Comment:
Why we should wrap the `get_node_inputs` output into the NumPy array? This method returns an iterable `list()`